### PR TITLE
Add the indexer configuration option

### PR DESCRIPTION
**Pull request by GGP1**
## Description

Adds the `indexer -> allow` value to the API configuration section of the documentation.

### Docs building
- [x] Compiles without warnings.

image

### Code formatting and web optimization
- [x] Uses three spaces indentation.
- [x] Adds or updates meta descriptions accordingly.
- [x] Updates the `redirects.js` script if necessary (check [this guide](https://github.com/wazuh/wazuh-documentation/blob/master/NEW_RELEASE.md)).
### Writing style
- [x] Uses present tense, active voice, and semi-formal registry.
- [x] Uses short, simple sentences.
- [x] Uses **bold** for user interface elements, _italics_ for key terms or emphasis, and `code` font for Bash commands, file names, REST paths, and code.

### CODE REVIEW

File: source/user-manual/api/configuration.rst [hunk 1]
```diff
@@ -347,6 +349,18 @@ agents
...
+| Sub-fields | Allowed values       | Default value | Description                                                                                                                                                                                                                                                                                                                                 |
++============+======================+===============+=============================================================================================================================================================================================================================================================================================================================================+
+| allow      | yes, true, no, false | true          | Allow uploading configurations with modified indexer values through the Wazuh API. Setting this option to ``false`` prevents uploading ``ossec.conf`` files if the ``<indexer>`` section has changed.                                                                                                                                       |
```
Comment (fdalmaup) [inline source/user-manual/api/configuration.rst:360]: I would add a link to the `indexer` tag reference (`user-manual/reference/ossec-conf/indexer.html`).

**Review by fdalmaup — CHANGES_REQUESTED** [review]

**Review by fdalmaup — DISMISSED** [review]
LGTM!

**Review by Selutario — APPROVED** [review]

File: source/user-manual/api/configuration.rst [hunk 2]
```diff
@@ -347,6 +349,18 @@ agents
...
+| Sub-fields | Allowed values       | Default value | Description                                                                                                                                                                                                                                                                                                                                 |
++============+======================+===============+=============================================================================================================================================================================================================================================================================================================================================+
+| allow      | yes, true, no, false | true          | Allow uploading configurations with modified indexer values through the Wazuh API. Setting this option to ``false`` prevents uploading ``ossec.conf`` files if the :ref:`indexer <reference_ossec_indexer>` section has changed.                                                                                                            |
```
Comment (javimed) [inline source/user-manual/api/configuration.rst:360]:
```suggestion
| allow      | yes, true, no, false | true          | Allows uploading an updated :doc:`indexer configuration section </user-manual/reference/ossec-conf/indexer>` through the Wazuh API. Setting this option to ``false`` prevents updating the indexer configuration when uploading ``ossec.conf``.                                                                                             |
```

File: source/user-manual/reference/ossec-conf/indexer.rst [hunk 3]
```diff
@@ -1,7 +1,9 @@
 .. Copyright (C) 2015, Wazuh, Inc.
 
 .. meta::
-  :description: 
+  :description:
+
```
Comment (javimed) [inline source/user-manual/reference/ossec-conf/indexer.rst:5]:
```suggestion
```

File: source/user-manual/reference/ossec-conf/indexer.rst [hunk 4]
```diff
@@ -1,7 +1,9 @@
...
+.. _reference_ossec_indexer:
```
Comment (javimed) [inline source/user-manual/reference/ossec-conf/indexer.rst:6]:
```suggestion
```

**Review by javimed — CHANGES_REQUESTED** [review]

**Review by javimed — APPROVED** [review]

**Pull request merged**